src/main/java/org/miracum/etl/fhirgateway/FhirGatewayApplication.java

```diff
@@ -2,10 +2,12 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.properties.ConfigurationPropertiesScan;
 import org.springframework.transaction.annotation.EnableTransactionManagement;
 
 @SpringBootApplication
 @EnableTransactionManagement
+@ConfigurationPropertiesScan
```

Comment:
Ich hab intensivst versucht zu verstehen, warum das nötig ist. Aber mehr als eine leise Ahnung hab ich nicht.
Ich versteh, dass der das inline property services.kafka.enabled in der Nested Test Class sonst nicht resolvet, aber warum nicht und sollte er das nicht trotzdem?

Reply:
da hab ich auch keine sinnvolle antwort: ich denke es ist notwendig weil record noch ein spezialfall in spring boot ist - zumindest wenn ich das nicht aktiv habe findet er die Konfigurationsklasse nicht. Das https://stackoverflow.com/a/74777176 oder `@EnableConfigurationProperties` müsste sonst beides gehen.